spec/unit/azurerm_server_delete_spec.rb

```diff
@@ -16,15 +16,15 @@
       Chef::Config[:knife][:azure_resource_group_name] = 'test-rg-group'
       @arm_server_instance.name_args = ['VM001']
 
-      @server_detail = double('server', :name => "VM001", :properties => double)
-      allow(@server_detail.properties).to receive_message_chain(:hardware_profile, :vm_size).and_return("10")
-      allow(@server_detail.properties).to receive_message_chain(:storage_profile, :os_disk, :os_type).and_return("Linux")
+      @server_detail = double('server', :name => "VM001", :hardware_profile => double, :storage_profile => double(:os_disk => double))
+      allow(@server_detail.hardware_profile).to receive_message_chain(:vm_size).and_return("10")
+      allow(@server_detail.storage_profile.os_disk).to receive_message_chain(:os_type).and_return("Linux")
```

Comment:
We shouldn't use `receive_message_chain` here since we are expecting a single method. Please use `receive` instead.